src/lib/vault/index.js

```diff
@@ -94,75 +98,120 @@ class Vault {
     _setSecret(key, value) {
         assert(key !== null && key !== undefined, `Cannot set key: [${key}]`);
         const path = `${this._secretMount}/${key}`;
-        return this._client.write(path, { value });
+        return this._client.write(path, value);
     }
 
     async _getSecret(key) {
         const path = `${this._secretMount}/${key}`;
-        const { data: { value } } = await this._client.read(path);
-        return value;
+        try {
+            const { data } = await this._client.read(path);
+            return data;
+        } catch (e) {
+            if (e.response && e.response.statusCode === 404) {
+                return null;
+            }
+            throw e;
+        }
     }
 
     async _deleteSecret(key) {
         const path = `${this._secretMount}/${key}`;
-        const { data } = await this._client.delete(path);
-        return data;
+        await this._client.delete(path);
+    }
+
+    async setClientCert(value) {
+        return this._setSecret(vaultPaths.CLIENT_CERT, value);
     }
 
-    async setClientPrivateKey(value) {
-        return this._setSecret(vaultPaths.CLIENT_PKEY, value);
+    async getClientCert() {
+        return this._getSecret(vaultPaths.CLIENT_CERT);
     }
 
-    async getClientPrivateKey() {
-        return this._getSecret(vaultPaths.CLIENT_PKEY);
+    async setJWS(value) {
+        return this._setSecret(vaultPaths.JWS, value);
     }
 
-    async setServerPrivateKey(value) {
-        return this._setSecret(vaultPaths.SERVER_PKEY, value);
+    async getJWS() {
+        return this._getSecret(vaultPaths.JWS);
     }
 
-    async getServerPrivateKey() {
-        return this._getSecret(vaultPaths.SERVER_PKEY);
+    async setPeerJWS(value) {
+        return this._setSecret(vaultPaths.PEER_JWS, value);
     }
 
-    async setJWSCerts(value) {
-        return this._setSecret(vaultPaths.JWS_CERTS, value);
+    async getPeerJWS() {
+        return this._getSecret(vaultPaths.PEER_JWS);
     }
 
-    async getJWSCerts() {
-        return this._getSecret(vaultPaths.JWS_CERTS);
+    async setILP(value) {
+        return this._setSecret(vaultPaths.ILP, value);
+    }
+
+    async getILP() {
+        return this._getSecret(vaultPaths.ILP);
     }
 
     /**
-     * co
-     * @param params
-     * @returns {Promise<any>}
+     * Delete root CA
+     * @returns {Promise<void>}
      */
-    async createCSR(params) {
-        const { parameters } = params;
+    async deleteCA () {
+        await this._client.request({
+            path: `/${this._pkiMount}/root`,
+            method: 'DELETE',
+        });
+    }
+
+    /**
+     * Create root CA
+     * @param {Object} caOptions
+     */
+    async createCA (caOptions) {
+        // eslint-disable-next-line no-empty
+        try { await this.deleteCA(); } catch (e) { }
+
+        const { names } = caOptions.csr;
+        const altNames = names.length > 1 ? names.slice(1).map((name) => name.CN).join(',') : '';
```

Comment:
_Might_ be clearer as:
```suggestion
        const [,...altNamesObjs] = names;
        const altNames = altNamesObjs.map((name) => name.CN).join(',');
```